.github/workflows/reusable-scorecard.yml

```diff
@@ -0,0 +1,64 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
+name: Scorecards supply-chain security
+on:
+  workflow_call:
+
+# Declare default permissions as none
+permissions: {}
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@e38b1902ae4f44df626f11ba0734b14fb91f8f86 # v2.1.2
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          # (Optional) "write" PAT token. Uncomment the `repo_token` line below if:
+          # - you want to enable the Branch-Protection check on a *public* repository, or
+          # - you are installing Scorecards on a *private* repository
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action#authentication-with-pat.
+          # repo_token: ${{ secrets.SCORECARD_TOKEN }}
```

Comment:
maybe make this an optional input to the workflow, so it can be set if desired? If someone wanted to do this, they'd have to create a copy of this which defeats the purpose of this being reusable IIUC

Reply:
Makes sense. Thanks for pointing it out!

Reply:
There is a "secrets: inherit" feature for GitHub reusable workflows. Does it make sense to use it here?
https://betterprogramming.pub/how-to-use-github-actions-reusable-workflow-8604e8cbf258

Reply:
only downside of that is that the secret names have to match. within this org that's probably OK, but not always true

Reply:
I added `scorecard_token` as workflow input as `secrets`. The `scorecard_token` is passed to Scorecard in `repo_token` input.

Example usage:
```yml
uses: ...
with:
  publish_results: true
secrets:
  scorecard_token: ${{ secrets.<your_secret_name> }}
```